prog/weave-npc/main.go

```diff
@@ -112,11 +137,93 @@ func createBaseRules(ipt *iptables.IPTables, ips ipset.Interface) error {
 	}
 
 	// If the destination address is not any of the local pods, let it through
-	if err := ips.Create(npc.LocalIpset, ipset.HashIP); err != nil {
+	found, err := ipsetExist(ips, npc.LocalIpset)
+	if err != nil {
+		return err
+	}
+	if !found {
+		if err := ips.Create(npc.LocalIpset, ipset.HashIP); err != nil {
+			return err
+		}
+	}
+	if err := ipt.Append(npc.TableFilter, npc.MainChain,
+		"-m", "set", "!", "--match-set", npc.LocalIpset, "dst", "-j", "ACCEPT"); err != nil {
 		return err
 	}
-	return ipt.Append(npc.TableFilter, npc.MainChain,
-		"-m", "set", "!", "--match-set", npc.LocalIpset, "dst", "-j", "ACCEPT")
+
+	if err := ipt.Append(npc.TableFilter, npc.EgressMarkChain,
+		"-j", "MARK", "--set-xmark", npc.EgressMark); err != nil {
+		return err
+	}
+
+	ruleSpecs := [][]string{
+		{"-m", "state", "--state", "RELATED,ESTABLISHED", "-j", "RETURN"},
+		{"-m", "state", "--state", "NEW", "-m", "set", "!", "--match-set", npc.LocalIpset, "src", "-j", "RETURN"},
```

Comment:
Unfortunately not - the fact that incoming interface is `weave` does not guarantee that a packet originated on a local node.

Reply:
If the packet  has arrived from the `weave` bridge interface, is it not sufficient condition to say packet is from one the pods running locally? 
Is `LocalIpset` inclusive of pods running in host network?

Reply:
> is it not sufficient condition to say packet is from one the pods running locally?

Unfortunately noup, because we connect a datapath to the `weave` bridge via `vethwe-bridge` link, so the bridge receives weave traffic from remote hosts as well.

> Is LocalIpset inclusive of pods running in host network?

It just includes a host local pods, but I think we can get rid of the ipset in favor of https://github.com/weaveworks/weave/issues/3344